spec/controllers/tags_controller_spec.rb

```diff
@@ -173,6 +173,45 @@
     end
   end
 
+  describe "show_hidden" do
+    let!(:fandom1) { create(:fandom, canonical: true) }
+    let!(:freeform1) { create(:freeform, canonical: false) }
+    let!(:character1) { create(:character, canonical: false) }
+    let!(:character2) { create(:character, canonical: false) }
+    let!(:work) do
+      create(:work,
+             posted: true,
+             fandom_string: fandom1.name,
+             character_string: "#{character1.name},#{character2.name}",
+             freeform_string: freeform1.name,
+             archive_warning_string: ArchiveConfig.WARNING_DEFAULT_TAG_NAME)
+    end
+
+    it "redirects to referer with an error for non-ajax warnings requests" do
+      referer = root_path
```

Comment:
The referer defaults to `root_path` if unset:

https://github.com/otwcode/otwarchive/blob/030fa4607d3c59fba2cacbe4e9c99b3c49fbc1cb/app/controllers/tags_controller.rb#L175

so it's better to use a different path to test it, e.g. `tags_path`.

Reply:
yep, sounds good! Resolved in 20ee3b511